modules/core/src/system.cpp

```diff
@@ -597,6 +599,10 @@ struct HWFeatures
         have[CV_CPU_VSX3] = (CV_VSX3);
     #endif
 
+    #ifdef __riscv
+        have[CV_CPU_RVV] = true;
```

Comment:
`__riscv` is very general and defined for general RISC-V architecture. There should be some specific check here.